### PR TITLE
Add Undo support for `CesiumGeoreference`, `CesiumSubScene`, and `CesiumGlobeAnchor`

### LINKED ISSUE #117: Implement georeference origin change "undo" in editor
In the editor, it would be useful to be able to Ctrl-Z undo a georeference origin change.

### PR DESCRIPTION
**Pull request by j9liu**
Fixes #117 and #104.

### CODE REVIEW

**Comment by j9liu** [comment]
Ended up using `OnValidate` for all of the classes. I think it's the most elegant solution and as long as these classes aren't handling object creation ([as noted in the documentation](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnValidate.html)), it should be fine. This now fixes #104 as well.

**Comment by j9liu** [comment]
@kring It looks like calling `Sync()` in `CesiumGlobeAnchor.OnValidate` is causing CI to fail because of the exception that it's not nested under a `CesiumGeoreference`. What's the best course of action -- should I put that line of code in a `try` block?

File: Editor/CesiumEditorUtility.cs [hunk 1]
```diff
@@ -325,7 +325,7 @@ public static CesiumSubScene CreateSubScene(CesiumGeoreference georeference)
                 georeference.ecefZ
             );
 
-            GameObject subSceneGameObject = new GameObject();
+            GameObject subSceneGameObject = new GameObject("New Sub-Scene");
```
Comment (j9liu) [inline Editor/CesiumEditorUtility.cs:328]: Small unrelated change that is in this PR for convenience.

**Comment by kring** [comment]
@j9liu this is causing a test failure in `SyncsUnityPositionFromTransformAndBackMultipleFrames` in `TestCesiumGlobeAnchor.cs`. (note to self: really need to get these tests, such as they are, running automatically on CI).

It seems that `OnValidate` is called immediately from `AddComponent`, so we can no longer observe the positionAuthority as "None" after adding the component. That's not a problem by itself, but it means there will be a difference in how the CesiumGlobeAnchor works in the Editor and in the Player:

* _In the player_: Adding the component does not immediately invoke Sync. So you can add the component and then immediately set the position, and this won't affect the orientation of the object.
* _In the editor_: Adding the component calls OnValidate, which "locks in" the position. If you then move it to the other side of the globe, its orientation will be flipped.

I'm not sure how big a problem this is. But fundamentally, the GlobeAnchor state transitions are ... really hard to get right. And it's even harder to be sure they're right. And I think letting a "called all the time, for lots of different reasons" method like OnValidate affect the GlobeAnchor's state makes it even harder to reason about.

So... is there a finer-grained alternative to OnValidate that would solve the initial problem here? If not, I'm sure we can make it work, but I think it's worth thinking about any alternatives.

**Comment by kring** [comment]
I added a new test for the scenario I described above. It's failing in this branch, but if you remove OnValidate, it passes.

**Comment by j9liu** [comment]
So while debugging, it looks like doing `OnValidate()` was wrong even for `CesiumGeoreference`, because when a script loads it calls `OnValidate` before `OnEnable`, so it was calling `UpdateOrigin` before `InitializeOrigin` :face_palm: It was actually causing the globe anchors to be disoriented, even without `OnvValidate()` in the globe anchor's code.

I might just give up and make all of these classes cache values that they compare their current values against. I have to do that for `CesiumGlobeAnchor` anyway and it would be nice to make it consistent across all related classes.

**Comment by kring** [comment]
This looks great, @j9liu. I like that only the Editor components have to deal with what amounts to an Editor-only problem, even if it's a bit of boilerplate.
